lighthouse-core/report/v2/report-renderer.js

```diff
@@ -25,19 +57,183 @@
  */
 window.ReportRenderer = class ReportRenderer {
   /**
-   * @param {!Object} report
+   * @param {Document} document
+   */
+  constructor(document) {
+    this._document = document;
+  }
+
+  /**
+   * @param {ReportJSON} report
+   * @return {Element}
+   */
+  renderReport(report) {
+    try {
+      return this._renderReport(report);
+    } catch (e) {
+      return this._renderException(e);
+    }
+  }
+
+  /**
+   * @param {string} name
+   * @param {string=} className
+   * @param {Object=} attrs Attribute key/val pairs.
+   * @returns {Element}
+   */
+  _createElement(name, className, attrs = {}) {
+    const element = this._document.createElement(name);
+    if (className) {
+      element.className = className;
+    }
+    for (const [key, val] of Object.entries(attrs)) {
```

Comment:
`/* eslint-env browser */` already present. im comfy with assuming this is only run clientside